### PR TITLE
Prepare 2023.01.16 release

### PR DESCRIPTION
**Pull request by freaz**
I believe it is time to release the specification. It also includes service specifier from far past and some fixes in specs.

### CODE REVIEW

**Review by TheEdward162 — CHANGES_REQUESTED** [review]
Something went very wrong here. This reverts most of the changes that were staged in draft. Check the diff on this PR, especially on `map-spec.md` (for example `OperationCallSlot` was fixed in draft but now is back to version from `latest`).

**Comment by freaz** [comment]
Somehow I have dejavu feeling. Hopefully I did copying correctly now.

**Review by TheEdward162 — APPROVED** [review]

File: spec/latest/map-spec.md [hunk 1]
```diff
@@ -218,13 +218,16 @@ set {
 
 # Operation Call
 
-OperationCall: `call` Iteration? OperationName OperationArguments? Condition? OperationCallSlot?
+OperationCall: `call` Iteration? OperationName OperationArguments Condition? { OperationCallSlot* }
 
-OperationArguments : ( Argument* )
+OperationArguments : ( Argument[comma]* )
```
[the remark this answers is not in the document]
Comment (zdne) [reply spec/latest/map-spec.md:223]: I also don't think this is the correct way to handle comma-separated lists per markdown spec. As `Argument[comma]*` seems to be saying `comma` is a [conditional parameter](https://spec-md.com/#sec-Conditional-Parameters).

I believe the correct way [could be](https://spec-md.com/#sec-Quantifiers.Use-with-Non-Terminals):

```
OperationArguments : ( ArgumentList* )

ArgumentList: ArgumentList `,`? Argument
```

Or, following the example set in the profile (see `ArrayLiteral` in profile.md):

```
OperationArguments : ( ArgumentList* )

ArgumentList : Argument ArgumentListContinued*

ArgumentListContinued : , Argument
```

**Review by zdne — COMMENTED** [review]
Caught some minor things with list of arguments. Otherwise looks good.

File: spec/latest/map-spec.md (hunk 1 above)
Comment (zdne) [inline spec/latest/map-spec.md:223]:
```suggestion
OperationArguments : ( ArgumentList* )

ArgumentList: ArgumentList `,`? Argument
```

File: spec/latest/common/definitions.md [hunk 2]
```diff
@@ -60,6 +60,10 @@ starwars/character-information@1.1
 
 ProviderIdentifier : DocumentNameIdentifier
 
+## Service Identifier
+
+ServiceIdentifier : Identifier
```
Comment (zdne) [inline spec/latest/common/definitions.md:65]:
```suggestion
ServiceIdentifier : Identifier

Service identifier form provider's definition. Generally used for specifying a service base (host) URL. 
```

**Review by zdne — APPROVED** [review]
Added suggestions to proceed with the PR

File: spec/latest/map-spec.md [hunk 3]
```diff
@@ -220,7 +220,9 @@ set {
 
 OperationCall: `call` Iteration? OperationName OperationArguments Condition? { OperationCallSlot* }
 
-OperationArguments : ( Argument[comma]* )
+OperationArguments : ( ArgumentList* )
+
+ArgumentList: ArgumentList `,`? Argument
```
Comment (TheEdward162) [inline spec/latest/map-spec.md:225]: I don't think this is valid @zdne, it's forever recursive.

In the link you posted (https://spec-md.com/#sec-Quantifiers.Use-with-Non-Terminals) they present have two branches, and the second one is always terminal (never recurses). Also the optional `|` is there to allow the "TypeScript" syntax with a leading `|`:
```ts
type MyUnion =
  | Foo
  | Bar;
```

Our rule never terminates. I suggest with go with your second suggestion, with an added optional trailing comma in `ArgumentList` (which is allowed in current parser):
```
OperationArguments : ( ArgumentList* )

ArgumentList : Argument ArgumentListContinued* `,`?

ArgumentListContinued : , Argument
```

**Review by TheEdward162 — CHANGES_REQUESTED** [review]

File: spec/latest/map-spec.md (hunk 3 above)
Comment (zdne) [reply spec/latest/map-spec.md]: Ok!

Comment (zdne) [reply spec/latest/map-spec.md]: Let's fix it and focus on other stuff :/

File: spec/latest/map-spec.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (freaz) [reply spec/latest/map-spec.md]: It should be fixed to my best knowledge.

**Pull request merged**